### PR TITLE
test: Increase timeout to mitigate non-deterministic test failure

### PR DESCRIPTION
**Pull request by Liamsi**
This should fix https://github.com/tendermint/tendermint/issues/3576 (ran it many times locally but only time will tell). The test actually only checked for the opcode of the error. From the name of the test we actually want to test if we see a timeout after a pre-defined time. 

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: privval/socket_listeners_test.go [hunk 1]
```diff
@@ -97,7 +97,15 @@ func TestListenerTimeoutAccept(t *testing.T) {
 }
 
 func TestListenerTimeoutReadWrite(t *testing.T) {
-	for _, tc := range listenerTestCases(t, time.Second, time.Millisecond) {
+	var (
```
Comment (melekes) [inline privval/socket_listeners_test.go]:
```suggestion
	const (
```

File: privval/socket_listeners_test.go [hunk 2]
```diff
@@ -122,5 +129,9 @@ func TestListenerTimeoutReadWrite(t *testing.T) {
 		if have, want := opErr.Op, "read"; have != want {
 			t.Errorf("for %s listener, have %v, want %v", tc.description, have, want)
 		}
+
+		if have, want := opErr.Timeout(), true; have != want {
+			t.Errorf("for %s listener, got unexpected error: have %v, want %v", tc.description, have, want)
```
Comment (melekes) [inline privval/socket_listeners_test.go]:
```suggestion
			t.Errorf("for %s listener, got unexpected error: have %v, want timeout error", tc.description, opErr)
```

File: privval/socket_listeners_test.go [hunk 3]
```diff
@@ -122,5 +129,9 @@ func TestListenerTimeoutReadWrite(t *testing.T) {
 		if have, want := opErr.Op, "read"; have != want {
 			t.Errorf("for %s listener, have %v, want %v", tc.description, have, want)
 		}
+
+		if have, want := opErr.Timeout(), true; have != want {
```
Comment (melekes) [inline privval/socket_listeners_test.go]:
```suggestion
		if !opErr.Timeout() {
```

Comment (Liamsi) [reply privval/socket_listeners_test.go]: I like to have this consistent with the cases above (would you also change to `if opErr.Op != "read"`)

Comment (melekes) [reply privval/socket_listeners_test.go]: Yes, I would

Comment (xla) [reply privval/socket_listeners_test.go]: The reason for this consistent form is that when you reformulate the expectations, you can't forget to change the error reprted. Less error prone and less editing.

Comment (melekes) [reply privval/socket_listeners_test.go]: `for %s listener, got unexpected error: have false want true`
in it's current form does not make sense for person running tests

Comment (melekes) [reply privval/socket_listeners_test.go]: @xla agree if you're talking about above case where we compare Op == read, but not here

Comment (xla) [reply privval/socket_listeners_test.go]: Agreed, it doesn't always produces the most readable output. A balance must be struck.

**Pull request merged**